File: patterns/c-patterns/pattern23.c
// Pattern 23: ...

// ABCDE
// ABCD*
// ABC**
// AB***
// A****

#include <stdio.h>
int main()
{
    int i, j;
    int n = 4;
    for (i = n; i >= 0; i--)
    {
        for (j = 0; j <= n; j++)
        {
            if (j > i)
                printf("*");
            else
                printf("%c", j + 65);
        }
        printf("\n");
    }
    return 0;
}

File: patterns/javascript-patterns/pattern23.js
// Pattern 23: ...

// ABCDE
// ABCD*
// ABC**
// AB***
// A****

const n = 4;

for (let i = n; i >= 0; i--) {
    let row = '';

    for (let j = 0; j <= n; j++) {
        if (j > i) {
            row += '*';
        } else {
            row += String.fromCharCode(j + 65);
        }
    }

    console.log(row);
}
